src/main/kotlin/ecommerce/service/ProductService.kt

```diff
@@ -0,0 +1,69 @@
+package ecommerce.service
+
+import ecommerce.dto.OptionResponse
+import ecommerce.dto.ProductRequest
+import ecommerce.exception.ConflictException
+import ecommerce.exception.NotFoundException
+import ecommerce.model.Product
+import ecommerce.model.mapper.OptionMapper
+import ecommerce.repository.ProductRepository
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import kotlin.math.min
+
+@Transactional
+@Service
+class ProductService(private val productRepository: ProductRepository) {
+    fun create(productRequest: ProductRequest): Long {
+        require(!productRequest.options.isNullOrEmpty()) { "options must not be empty" }
+
+        if (productRepository.existsByName(productRequest.name)) {
+            throw ConflictException("Product with name ${productRequest.name} already exists")
+        }
+        val product = productRepository.save(productRequest.toProduct())
+        return product.id
+            ?: throw NotFoundException("Product with name ${productRequest.name} not found")
+    }
+
+    fun read(): List<Product> {
```

Comment:
```suggestion
    @Transactional(readOnly = true)
    fun read(): List<Product> {
```

You could consider setting `readOnly` when you're only querying data 👀

Reply:
Thanks for letting me know about the feature. I applied it.